### PR TITLE
Allow to configure `SerializeImagePulls` for kubelet configuration in `Shoot` API

### LINKED ISSUE #4855: Make kubelet parameter `--serialize-image-pulls` configurable
**How to categorize this issue?**

/area control-plane
/kind enhancement
/priority 3

**What would you like to be added**:

In a shoot manifest some kubelet parameters can already be configured via `spec.kubernetes.kubelet.*`.

There should be an additional field `serializeImagePulls` with boolean value that directly translates into kubelet parameter `--serialize-image-pulls`.

**Why is this needed**:

Pulling images in parallel improves pod startup performance.

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area storage scalability
/kind enhancement

**What this PR does / why we need it**:

Allow to configure `SerializeImagePulls` for kubelet configuration in `Shoot` API

**Which issue(s) this PR fixes**:
Fixes #4855

**Special notes for your reviewer**:
CC: @rfranzke 

**Release note**:
```other operator
It's now possible to configure the `SerializeImagePulls` fields for the kubelet configuration (defaults: true) in the `Shoot` API via `.spec.{provider.workers[]}.kubernetes.kubelet.SerializeImagePulls`.
```

### CODE REVIEW

File: pkg/apis/core/v1alpha1/defaults_test.go [hunk 1]
```diff
@@ -348,6 +348,22 @@ var _ = Describe("Defaults", func() {
 			Expect(obj.Spec.Kubernetes.Kubelet.ImageGCLowThresholdPercent).To(PointTo(Equal(low)))
 		})
 
+		It("should default the serializeImagePulls fields", func() {
+			SetDefaults_Shoot(obj)
+
+			Expect(obj.Spec.Kubernetes.Kubelet.SerializeImagePulls).To(PointTo(BeTrue()))
+		})
+
+		It("should not default the serializeImagePulls field", func() {
```
Comment (plkokanov) [inline pkg/apis/core/v1alpha1/defaults_test.go]:
```suggestion
		It("should not default the serializeImagePulls field if it is already set", func() {
```

File: pkg/apis/core/v1beta1/defaults_test.go [hunk 2]
```diff
@@ -348,6 +348,22 @@ var _ = Describe("Defaults", func() {
 			Expect(obj.Spec.Kubernetes.Kubelet.ImageGCLowThresholdPercent).To(PointTo(Equal(low)))
 		})
 
+		It("should default the serializeImagePulls fields", func() {
+			SetDefaults_Shoot(obj)
+
+			Expect(obj.Spec.Kubernetes.Kubelet.SerializeImagePulls).To(PointTo(BeTrue()))
+		})
+
+		It("should not default the serializeImagePulls field", func() {
```
Comment (plkokanov) [inline pkg/apis/core/v1beta1/defaults_test.go]:
```suggestion
		It("should not default the serializeImagePulls field if it is already set", func() {
```

**Review by plkokanov — COMMENTED** [review]
Thanks for the PR! 
Just a small nit.

File: pkg/apis/core/types_shoot.go [hunk 3]
```diff
@@ -663,6 +663,8 @@ type KubeletConfig struct {
 	ImageGCHighThresholdPercent *int32
 	// ImageGCLowThresholdPercent describes the percent of the disk to which garbage collection attempts to free.
 	ImageGCLowThresholdPercent *int32
+	//SerializeImagePulls describes whether the images are pulled one at a time.
```
Comment (ialidzhikov) [inline pkg/apis/core/types_shoot.go]:
```suggestion
	// SerializeImagePulls describes whether the images are pulled one at a time.
```

File: pkg/apis/core/v1alpha1/types_shoot.go [hunk 4]
```diff
@@ -836,6 +836,10 @@ type KubeletConfig struct {
 	// +optional
 	// Default: 40
 	ImageGCLowThresholdPercent *int32 `json:"imageGCLowThresholdPercent,omitempty" protobuf:"bytes,17,opt,name=imageGCLowThresholdPercent"`
+	//SerializeImagePulls describes whether the images are pulled one at a time.
```
Comment (ialidzhikov) [inline pkg/apis/core/v1alpha1/types_shoot.go]:
```suggestion
	// SerializeImagePulls describes whether the images are pulled one at a time.
```

File: pkg/apis/core/v1alpha1/defaults_test.go [hunk 5]
```diff
@@ -348,6 +348,22 @@ var _ = Describe("Defaults", func() {
 			Expect(obj.Spec.Kubernetes.Kubelet.ImageGCLowThresholdPercent).To(PointTo(Equal(low)))
 		})
 
+		It("should default the serializeImagePulls fields", func() {
```
Comment (ialidzhikov) [inline pkg/apis/core/v1alpha1/defaults_test.go]:
```suggestion
		It("should default the serializeImagePulls field", func() {
```

File: pkg/apis/core/v1beta1/defaults_test.go [hunk 6]
```diff
@@ -348,6 +348,22 @@ var _ = Describe("Defaults", func() {
 			Expect(obj.Spec.Kubernetes.Kubelet.ImageGCLowThresholdPercent).To(PointTo(Equal(low)))
 		})
 
+		It("should default the serializeImagePulls fields", func() {
```
Comment (ialidzhikov) [inline pkg/apis/core/v1beta1/defaults_test.go]:
```suggestion
		It("should default the serializeImagePulls field", func() {
```

File: pkg/apis/core/v1beta1/types_shoot.go [hunk 7]
```diff
@@ -848,6 +848,10 @@ type KubeletConfig struct {
 	// +optional
 	// Default: 40
 	ImageGCLowThresholdPercent *int32 `json:"imageGCLowThresholdPercent,omitempty" protobuf:"bytes,17,opt,name=imageGCLowThresholdPercent"`
+	//SerializeImagePulls describes whether the images are pulled one at a time.
```
Comment (ialidzhikov) [inline pkg/apis/core/v1beta1/types_shoot.go]:
```suggestion
	// SerializeImagePulls describes whether the images are pulled one at a time.
```

**Review by ialidzhikov — COMMENTED** [review]
Minor styling suggestions, otherwise lgtm.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**